pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	zconfig "github.com/lf-edge/eve/api/go/config"
+	"github.com/lf-edge/eve/api/go/profile"
+	"github.com/lf-edge/eve/pkg/pillar/flextimer"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+const (
+	defaultLocalProfileServerPort = "8888"
+	savedLocalProfileFile         = "lastlocalprofile"
+)
+
+// urlAndSrcIP structure for mapping source IP and url of local server
+type urlAndSrcIP struct {
+	srcIP     net.IP
+	actualURL string
+}
+
+func getLocalProfileURL(localProfileServer string) (string, error) {
+	localProfileURL := fmt.Sprintf("http://%s", localProfileServer)
+	u, err := url.Parse(localProfileURL)
+	if err != nil {
+		return "", fmt.Errorf("url.Parse: %s", err)
+	}
+	if u.Port() == "" {
+		localProfileURL = fmt.Sprintf("%s:%s", localProfileURL, defaultLocalProfileServerPort)
+	}
+	return fmt.Sprintf("%s/api/v1/local_profile", localProfileURL), nil
+}
+
+//checkAndHandleLocalProfileUpdate gets local profile config if localProfileServer defined,
+//determines currentProfile and notify on changes
+func checkAndHandleLocalProfileUpdate(getconfigCtx *getconfigContext) {
+	localProfileServer := getconfigCtx.localProfileServer
+	if localProfileServer == "" {
+		return
```

Comment:
Need to handle the case when getconfigCtx.localProfile is set, since that indicates that the localProfileServer was just cleared. In that case, need to clear the saved file, check if currentProfile will be updated to to clearling localProfile, and log the change.

Reply:
It is here: https://github.com/lf-edge/eve/pull/2105/files#diff-d48cdc10ee2eb48b440bb875352eb01e2fc67a5927863c307628c19476028b6cR293